### PR TITLE
[test] remove Java source level 1.5 workaround

### PR DESCRIPTION
**Pull request by jukzi**
to avoid warnings in build output

https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2975

### CODE REVIEW

**Comment by jukzi** [comment]
removed 352 silly warning about VerifyTests.java from consoleText.txt

**Comment by stephan-herrmann** [comment]
To keep our code base consistent, the javadoc of org.eclipse.jdt.core.tests.util.VerifyTests needs an update, too. There may be more in that class.

**Comment by srikanth-sankaran** [comment]
> removed 352 silly warning about VerifyTests.java from consoleText.txt

Thanks @jukzi 

1. I am a little bit unclear about where we stand wrt to this PR. The associated issue https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2975 as per the top comment [here](https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2976#issue-2532974103) appears closed already as a false alert ?? Was that wrong linking ? 

2. Could you also share a link to the console output log so I can peruse what 352 warnings you are referring to

3. Did you want to look into @stephan-herrmann's comment above as a part of this PR and if so want to provide a further commit  ?

**Comment by jukzi** [comment]
https://download.eclipse.org/eclipse/downloads/drops4/I20240917-1800/testresults/consolelogs/ep434I-unit-win32-java17_win32.win32.x86_64_17_consolelog.txt
Contains a lot of warning that distract from real issues:
image
I will update this PR to Stephan's comments.

File: org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/util/VerifyTests.java [hunk 1]
```diff
@@ -34,12 +34,8 @@
  * changes in this class work as expected, without the need to update the hard-coded default value every single time
  * during an ongoing refactoring.
  * <p>
- * In order to make the copying job easier, keep this class compatible with Java 5 language level. You may however use
- * things like {@code @Override} for interfaces, {@code assert} (if in a single line), {@code @SuppressWarnings},
- * because {@link TestVerifier#getVerifyTestsCode()} can filter them out dynamically. You should however avoid things
- * like diamonds, multi-catch, catch-with-resources and more recent Java features.
+ * In order to make the copying job easier, keep this class compatible with the lowest supported Java language level (1.8).
```
Comment (srikanth-sankaran) [inline org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/util/VerifyTests.java:37]: The hardcoding  of 1.8 could have been avoided, but is not a big deal.

**Review by srikanth-sankaran — COMMENTED** [review]
Changes look good, sorry you beat me to it by 15 minutes. Thanks for doing this @jukzi

**Pull request merged**